lib/scheduler.js

```diff
@@ -0,0 +1,225 @@
+'use strict';
+
+// const debug = console.log;
+const debug = function() {};
+const info = console.log;
+
+const Combinator = require('./combinator.js');
+
+// TODO: Load teams from a file.
+const teams = [
+  { key: 'A', schedule: [], history: [] },
+  { key: 'B', schedule: [], history: [] },
+  { key: 'C', schedule: [], history: [] },
+  { key: 'D', schedule: [], history: [] },
+
+  { key: 'E', schedule: [], history: [] },
+  { key: 'F', schedule: [], history: [] },
+  { key: 'G', schedule: [], history: [] },
+  { key: 'H', schedule: [], history: [] },
+
+  { key: 'I', schedule: [], history: [] },
+  { key: 'J', schedule: [], history: [] },
+  { key: 'K', schedule: [], history: [] },
+  { key: 'L', schedule: [], history: [] },
+
+  { key: 'M', schedule: [], history: [] },
+  { key: 'N', schedule: [], history: [] },
+  { key: 'O', schedule: [], history: [] },
+  { key: 'P', schedule: [], history: [] },
+];
+
+const machines = [];
+for(let i = 0; i < 20; i++) {
+  machines.push(`M${i+1}`);
+}
+
+debug('MACHINES:', machines);
+
+const lookup = {};
+teams.forEach(team => lookup[team.key] = team);
+
+debug('LOOKUP:', lookup);
+
+const numRounds = 3;
+debug('numRounds:', numRounds);
+const MAX_HA = Math.ceil(numRounds / 2);
+debug('MAX_HA:', MAX_HA);
+
+function getHomeAway(team) {
+  let nHome = 0, nAway = 0, streak = 0;
+  team.schedule.forEach(match => {
+    const isHome = match.home === team.key;
+    // debug('isHome', isHome, match);
+    if(isHome) {
+      nHome++;
+      if(streak < 0) streak = 0;
+      streak++;
+    }
+    if(!isHome) {
```

Comment:
Brian told us never to use else. Although, he also said that none of the rules were absolute. ;)